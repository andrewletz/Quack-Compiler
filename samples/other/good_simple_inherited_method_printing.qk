class C1 () {
    def STR() : String {
        return "This is a C1.\n";
    }
}

class C2() extends C1 {
    def STR() : String {
        return "This is a C2.\n";
    }
}

my_obj : Obj = C1();
my_obj.PRINT();

my_obj = C2();
my_obj.PRINT();

