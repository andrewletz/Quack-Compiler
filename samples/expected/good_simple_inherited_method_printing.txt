This is a C1.
This is a C2.
